src/pages/home/uploadSong/UploadSong.tsx

```diff
@@ -179,15 +181,24 @@ const UploadSong: FunctionComponent = () => {
         return languageCodes.includes(countryCode);
       }),
     barcodeType: Yup.string(),
-    barcodeNumber: Yup.string().when("barcodeType", {
-      is: (barcodeType: string) => !!barcodeType && barcodeType !== NONE_OPTION,
-      then: Yup.string().required(
-        "Barcode number is required when barcode type is selected"
-      ),
-      otherwise: Yup.string(),
-    }),
+    barcodeNumber: Yup.string()
+      .max(
+        MAX_CHARACTER_COUNT,
+        `Must be ${MAX_CHARACTER_COUNT} characters or less`
```

Comment:
Same as `IPI` and `ISWC`, `barcodeNumber` (based on barcode type) would have specific regex validation and not just a character count check?